### PR TITLE
refactor: uniform kratos endpoints

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- A kratos library that has 3 major functions, generic ones to initialize, get, and submit flow.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

File: packages/shared/src/contexts/AuthContext.tsx [hunk 1]
```diff
@@ -69,7 +70,14 @@ const appendLogoutParam = (link: string): URL => {
...
 const logout = async (logoutUrl: string, token: string): Promise<void> => {
-  await Promise.all([dispatchLogout(), logoutSession(logoutUrl, token)]); // temporary
+  await Promise.all([
+    dispatchLogout(),
+    submitKratosFlow({
+      action: logoutUrl,
+      params: { csrf_token: token },
+      method: 'GET',
+    }),
+  ]); // temporary
```
Comment (rebelchris) [inline packages/shared/src/contexts/AuthContext.tsx:80]: The API logout already handles Kratos logout, no need to extra call it.

File: packages/shared/src/lib/kratos.ts [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+export enum AuthFlow {
+  Login = '/login',
+  Logout = '/logout',
+  Registration = '/registration',
+  Settings = '/settings',
+  Recovery = '/recovery',
+  Verification = '/verification',
+}
+
+const authUrl = process.env.NEXT_PUBLIC_AUTH_URL || 'http://127.0.0.1:4433';
```
Comment (rebelchris) [inline packages/shared/src/lib/kratos.ts:158]: We probably want to move this away to a const file

File: packages/shared/src/lib/kratos.ts [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+}
+
+export interface SuccessfulRegistrationData {
+  session: AuthSession;
+  identity: Identity;
+}
+
+export enum AuthFlow {
+  Login = '/login',
+  Logout = '/logout',
```
Comment (rebelchris) [inline packages/shared/src/lib/kratos.ts:151]: Don't think we need a separate flow for this.

File: packages/shared/src/lib/kratos.ts [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+  Registration = '/registration',
+  Settings = '/settings',
```
Comment (rebelchris) [inline packages/shared/src/lib/kratos.ts:153]: What falls under settings flow?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Nice work Lee!
Just some questions around things we probably no longer need.

File: packages/shared/src/lib/kratos.ts (hunk 4 above)
Comment (sshanzel) [reply packages/shared/src/lib/kratos.ts:153]: The reset password should be under it, it is technically a settings flow.

File: packages/shared/src/contexts/AuthContext.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/contexts/AuthContext.tsx:80]: Awesome. We should remove this then. Thank you for the confirmation.

File: packages/shared/src/lib/kratos.ts (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/lib/kratos.ts:158]: Totally agree.

**Review by rebelchris — APPROVED** [review]
Awesome! 💖

**Pull request merged**